File: Java version of Life Game/Life_Game/Life_Game/Actions.java
package Main.Life_Game;

import java.util.ArrayList;

public class Actions {
    private static boolean game_over_E=false; // trigger to shut down the game
    private static boolean game_over_H=false; // trigger to shut down the game
    private static boolean game_over_C=false; // trigger to shut down the game
    private static int iteration=0; // basically time counter

    public static int getIteration() {
        return iteration;
    }
    public static void move_everybody() {
        for (Herbivore hb:
             Game_field.getHerbivores()) {
            hb.move();
        }
        for (Carnivore cv:
             Game_field.getCarnivores()) {
            cv.move();
        }
        iteration++; //one month mb
    }

    public static void eat_everybody(){
        ArrayList<Herbivore> herbivores = Game_field.getHerbivores();
        ArrayList<Carnivore> carnivores = Game_field.getCarnivores();

        for (Herbivore hb:
                herbivores) {
            hb.eat();
        }
        for(Carnivore cv:
                carnivores){
            cv.eat();
        }
    }

    public static void refresh(){
        ArrayList<Herbivore> herbivores = Game_field.getHerbivores();
        ArrayList<Carnivore> carnivores = Game_field.getCarnivores();

        if (herbivores.size()>=1){
            herbivores.removeIf(hb -> !hb.is_alive()); // Collections.removeIf!!!
        }
        if (carnivores.size()>=1){
            carnivores.removeIf(cv -> !cv.is_alive());
        }

        if (herbivores.size()==0 && carnivores.size()==0){
            game_over_Everybody();
        }
        if(herbivores.size()==0){
            game_over_Herbivores();
        }

        if(carnivores.size()==0){
            game_over_Carnivores();
        }
    }

    public static void game_over_Everybody(){
        game_over_E=true;
    }
    public static void game_over_Herbivores(){
        game_over_H=true;
    }
    public static void game_over_Carnivores(){
        game_over_C=true;
    }

    public static boolean isGame_over_E() {
        return game_over_E;
    }

    public static boolean isGame_over_C() {
        return game_over_C;
    }

    public static boolean isGame_over_H() {
        return game_over_H;
    }
}

File: newVersion/Actions.cpp

#include "Actions.h"
#include "Game_field.h"
#include "Carnivore.h"
#include "Herbivore.h"
#include <algorithm>
bool Actions::game_over_E = false;
bool Actions::game_over_H = false;
bool Actions::game_over_C = false;
int Actions::iteration = 0;

int Actions::getIteration() {
    return iteration;
}
bool Actions::isGameOverE() {
    return game_over_E;
}

bool Actions::isGameOverC() {
    return game_over_C;
}

bool Actions::isGameOverH() {
    return game_over_H;
}


void Actions::moveEverybody() {
    for (Herbivore hb : Game_field::getHerbivores()) {
        hb.move();
    }

    for (Carnivore cv : Game_field::getCarnivores()) {
        cv.move();
    }

    iteration++;
}

void Actions::eatEverybody() {
    for (Herbivore hb : Game_field::getHerbivores()) {
        hb.eat();
    }

    for (Carnivore cv : Game_field::getCarnivores()) {
        cv.eat();
    }
}

void Actions::refresh() {
    std::vector<Herbivore> herbivores = Game_field::getHerbivores();
    std::vector<Carnivore> carnivores = Game_field::getCarnivores();

    /*
    auto herbivorePredicate = [](const Herbivore& hb) { return !hb.isAliveStatus(); };
    herbivores.erase(std::remove_if(herbivores.begin(), herbivores.end(), herbivorePredicate),
                     herbivores.end());

    auto carnivorePredicate = [](const Carnivore& cv) { return !cv.isAliveStatus(); };
    carnivores.erase(std::remove_if(carnivores.begin(), carnivores.end(), carnivorePredicate),
                     carnivores.end());
    */

    if (herbivores.empty() && carnivores.empty()) {
        gameOverEverybody();
    }

    if (herbivores.empty()) {
        gameOverHerbivores();
    }

    if (carnivores.empty()) {
        gameOverCarnivores();
    }
}

void Actions::gameOverEverybody() {
    game_over_E = true;
}

void Actions::gameOverHerbivores() {
    game_over_H = true;
}

void Actions::gameOverCarnivores() {
    game_over_C = true;
}


